### ISSUE TITLE
Certain sequences of flags will cause glitches in wml loading (GNA #24705)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1557]
**[Original submission](https://gna.org/bugs/index.php?24705)** by sigurdfdragon on 2016-05-25

Certain sequences of flags will cause glitches in wml loading.

I haven't figured out how exactly to reproduce this yet, but some sequence of opening and closing wesnoth using combinations of '--no-addon', '--debug', and no flags will cause the PYR Add-on to report a wml error when loading.

Next time I catch this, I'll upload a sceenshot of the error

A cache purge fixes this when it occurs.

(Reproduced on Win 7 x64)
Release: 1.13.4+dev
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by sigurdfdragon** [comment]
@Wedge009 not really, but it might be related/be the known issue of cache corruption listed here https://github.com/wesnoth/wesnoth/issues/1924

**Comment by Wedge009** [comment]
Hmm, could be the same thing. @singalen noted there that he gets it with the AI Demos add-on. I don't have any add-ons except for a handful of campaigns in 1.14 and your Debug Tools.

(BTW, will you think you be publishing your debug tools for 1.15 as well, please?)

**Comment by sigurdfdragon** [comment]
@Wedge009 I will at some point. The 1.14 version should work as is on 1.15

**Comment by Pentarctagon** [comment]
Coming back to this - has any new information come up?  If nothing new has come up since and it's suspected to be related to another bug, I'd be in favor of closing this one.

**Comment by Wedge009** [comment]
@sigurdfdragon mentioned #1924 and that's still open.

**Comment by stevecotton** [comment]
I'd assume it's a different issue that sometimes means no add-ons appear, and I'm not sure if there's a clearly-written report of that one. I get it sometimes, but still feel I'm missing a part of what triggers it; if it was exactly what I think then it would be easily reproducible, which it isn't. Locally I run the unit-tests (and a subset of the schema tests) after each build, which would have the cache built with `--noaddons` each time.

The cache records the values of the preprocessor flags, but doesn't note whether `--noaddons` was in effect. So generating the cache with `--noaddons` can make subsequent runs of Wesnoth not see any add-ons; that state continues until you (purge the cache and then quit before regenerating the cache).
